pkg/plugins/builtin/script/script.go

```diff
@@ -0,0 +1,142 @@
+package script
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	gexec "os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/ovh/utask"
+
+	"github.com/ovh/utask/pkg/plugins/taskplugin"
+)
+
+// the script plugin execute scripts
+var (
+	Plugin = taskplugin.New("script", "0.1", exec,
+		taskplugin.WithConfig(validConfig, Config{}),
+	)
+)
+
+// Metadata represents the metadata of script execution
+type Metadata struct {
+	ExitCode      string `json:"exit_code"`
+	ProcessState  string `json:"process_state"`
+	Output        string `json:"output"`
+	ExecutionTime string `json:"execution_time"`
+	Error         string `json:"error"`
+}
+
+// Config is the configuration needed to execute a script
+type Config struct {
+	File    string   `json:"file,required"`
```

Comment:
Another proposition:

not sure about `file`. It could be `command`. And i would be able to set it this way:
```
command: python myScript.py arg1 arg2 arg3
```
or
```
command: bash -C "cd myScript && pip install && python main.py arg1 arg2 arg3"
```
To run it we have to split this string, consider slice[0] as the command and slice[1:] its arguments .
This way we juste have to check that the command valid: https://golang.org/pkg/os/exec/#LookPath

Reply:
Why split ? Never try to split on " " to execute a command, it will fail.

It should be written that way:
command: [ "bash", "-c", "cd myScript && pip install && python main.py arg1 arg2 arg3"]

Reply:
@fsamin I think your example perfectly shows why it should be _file_ and not an arbitrary shell command: we want to treat these scripts as black boxes with as little logic as possible in the step configuration.
The setup should be done elsewhere (which interpreter = shebang, pip install = definitely not at each step exec, but one-shot at Dockerfile?).
The current directory is interesting and may warrant a separate, generic config field.

Reply:
Checked /w @fsamin.